### PR TITLE
Run docker fake intake test with new version of the fakeintake if mod…

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Change test-infra-definitions pre test to use the newly built fakeintake if needed.

### Motivation

Avoid blind merge of PR modifying fake intake

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -763,6 +763,26 @@ workflow:
 
 # New E2E related rules
 
+.on_e2e_or_fakeintake_changes_or_manual:
```
Comment (pducolin) [inline .gitlab-ci.yml:766]: **💬 suggestion**
Should go to https://github.com/DataDog/datadog-agent/blob/pducolin/ADXT-124-update-qa-labels-doc/.gitlab/dev_container_deploy/fakeintake.yml#L2-L15

Comment (pducolin) [reply .gitlab-ci.yml:766]: Nevermind

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -763,6 +763,26 @@ workflow:
...
+  - <<: *if_main_branch
+  - !reference [.except_mergequeue]
+  - changes:
+      paths:
+        - test/new-e2e/fakeintake/**/*
+      compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
+    variables:
+      FAKEINTAKE_IMAGE_OVERRIDE: "public.ecr.aws/datadog/fakeintake:v$CI_COMMIT_SHORT_SHA"
```
Comment (pducolin) [inline .gitlab-ci.yml:774]: **💬 suggestion**
Would move this to https://github.com/DataDog/datadog-agent/blob/main/.gitlab/dev_container_deploy/fakeintake.yml#L15 to keep publishing and target image in the same context

Comment (KevinFairise2) [reply .gitlab-ci.yml:772]: We used to define all the gitlab rules in `.gitlab-ci.yaml` I think that's better for readability. If I move it I cannot reuse the yaml `*fakeintake_changes` anchor, which would duplicate some codes

**Review by amenasria — APPROVED** [review]
LGTM !

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -993,22 +1019,11 @@ workflow:
...
-  - <<: *on_fakeintake_changes
-    when: manual
-    allow_failure: true
+.on_fakeintake_changes:
+  - <<: *fakeintake_changes
 
 .on_fakeintake_changes_on_main:
-  - <<: *on_fakeintake_changes
+  - <<: *fakeintake_changes
     if: $CI_COMMIT_BRANCH == "main"
```
Comment (chouetz) [inline .gitlab-ci.yml:1027]: nitpick
```suggestion
  - <<: *if_main_branch
```

**Review by chouetz — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**